File: Source/Winforms.Extended/ColorProgressBar.cs
using System.Drawing;

namespace System.Windows.Forms
{
    public class ColorProgressBar : Control
    {
        private Brush _backBrush;
        private Pen _borderPen;
        private Color _borderColor = Color.DarkSlateGray;
        private Brush _foreBrush;
        private int _min = 0;
        private int _max = 100;
        private int _val = 0;
        public int Step { get; set; } = 10;

        public override Color BackColor
        {
            get => base.BackColor;
            set
            {
                base.BackColor = value;
                _backBrush = new SolidBrush(value);
                Invalidate();
            }
        }

        public Color BorderColor
        {
            get => _borderColor;
            set
            {
                _borderColor = value;
                _borderPen = new Pen(value);
                Invalidate();
            }
        }

        public override Color ForeColor
        {
            get => base.ForeColor;
            set
            {
                base.ForeColor = value;
                _foreBrush = new SolidBrush(value);
                Invalidate();
            }
        }

        public int Minimum
        {
            get => _min;
            set
            {
                if (_val < value) _val = value;
                _min = value;
                Invalidate();
            }
        }

        public int Maximum
        {
            get => _max;
            set
            {
                if (_val > value) _val = value;
                _max = value;
                Invalidate();
            }
        }

        public int Value
        {
            get => _val;
            set
            {
                if (value < _min) _val = _min;
                else if (value > _max) _val = _max;
                else _val = value;
                Invalidate();
            }
        }

        public void PerformStep()
        {
            Value += Step;
        }

        public void PerformStep(int amount)
        {
            Value += amount;
        }

        protected override void OnCreateControl()
        {
            if (_backBrush == null) _backBrush = new SolidBrush(base.BackColor);
            if (_borderPen == null) _borderPen = new Pen(_borderColor);
            if (_foreBrush == null) _foreBrush = new SolidBrush(base.ForeColor);
            base.OnCreateControl();
        }

        protected override void OnPaint(PaintEventArgs e)
        {
            var rect = e.ClipRectangle;
            rect.Width -= 1; rect.Height -= 1;
            e.Graphics.FillRectangle(_backBrush, rect);
            e.Graphics.DrawRectangle(_borderPen, rect);
            rect = new Rectangle(1, 1, rect.Width - 1, rect.Height - 1);

            e.Graphics.FillRectangle(_foreBrush, 1, 1,
                System.Convert.ToInt32(rect.Width *
                (System.Convert.ToDouble(Value / (double)Maximum))),
                rect.Height);
        }
    }
}

File: source/Launcher/My Project/Resources/User Controls/ColorProgressBar.vb
Friend Class ColorProgressBar : Inherits Control

    Private _backBrush As Brush
    Private _borderPen As Pen
    Private _borderColor As Color = Color.DarkSlateGray
    Private _foreBrush As Brush
    Private _min As Integer = 0
    Private _max As Integer = 100
    Private _val As Integer = 0
    Public Property StepCount As Integer = 10

    Public Overrides Property BackColor As Color
        Get
            Return MyBase.BackColor
        End Get
        Set
            MyBase.BackColor = Value
            _backBrush = New SolidBrush(Value)
            Invalidate()
        End Set
    End Property

    Public Property BorderColor As Color
        Get
            Return _borderColor
        End Get
        Set
            _borderColor = Value
            _borderPen = New Pen(Value)
            Invalidate()
        End Set
    End Property

    Public Overrides Property ForeColor As Color
        Get
            Return MyBase.ForeColor
        End Get
        Set
            MyBase.ForeColor = Value
            _foreBrush = New SolidBrush(Value)
            Invalidate()
        End Set
    End Property

    Public Property Minimum As Integer
        Get
            Return _min
        End Get
        Set
            If _val < Value Then _val = Value
            _min = Value
            Invalidate()
        End Set
    End Property

    Public Property Maximum As Integer
        Get
            Return _max
        End Get
        Set
            If _val > Value Then _val = Value
            _max = Value
            Invalidate()
        End Set
    End Property

    Public Property Value As Integer
        Get
            Return _val
        End Get
        Set
            If Value < _min Then
                _val = _min
            ElseIf Value > _max Then
                _val = _max
            Else
                _val = Value
            End If
            Invalidate()
        End Set
    End Property

    Public Sub PerformStep()
        Value += StepCount
    End Sub

    Public Sub PerformStep(amount As Integer)
        Value += amount
    End Sub

    Protected Overrides Sub OnCreateControl()
        If _backBrush Is Nothing Then
            _backBrush = New SolidBrush(MyBase.BackColor)
        End If

        If _borderPen Is Nothing Then
            _borderPen = New Pen(_borderColor)
        End If

        If _foreBrush Is Nothing Then
            _foreBrush = New SolidBrush(MyBase.ForeColor)
        End If

        MyBase.OnCreateControl()
    End Sub

    Protected Overrides Sub OnPaint(e As PaintEventArgs)
        Dim rect = e.ClipRectangle
        rect.Width -= 1
        rect.Height -= 1

        e.Graphics.FillRectangle(_backBrush, rect)
        e.Graphics.DrawRectangle(_borderPen, rect)

        rect = New Rectangle(1, 1, rect.Width - 1, rect.Height - 1)

        e.Graphics.FillRectangle(_foreBrush, 1, 1,
            CInt(rect.Width * (Value / Maximum)), rect.Height)
    End Sub
End Class